### PR TITLE
fix: #48815 - Upper left search icon always visible

### PR DESCRIPTION
**Pull request by CsabaTheKiss**
## Prerequisites

Please make sure you can check the following boxes:

-   [x] My code follows the code style of this project
-   [x] All new and existing tests passed
-   [x] Fix any eslint/prettier warnings/errors: npm run lint
-   [x] Remove unnecessary console.log (use logInfo logWarn if it's needed)
-   [x] Are variable and function names properly describing what it does?

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

-   [x] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
-   [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] I have updated the documentation accordingly
-   [ ] I have added tests to cover my changes

### Description
**Issue:**
Search panel icon is visible on the main page.
Did some digging: previously the visibility of the search icon was driven based on the visited path. Planning to make this check based on this again. Need to do some updates in our infrastructure first, though.

**Solution:**
Making possible to hide a toggle panel icon button. Would render the given panel's toggle button only if `isToggleButtonVisible` is `true`.  Can be function or a boolean primitive value.

The corePanelLeft and corePanelRight components would be updated: the toggle buttons would be rendered conditionally.

What was done:
- Extending Panel interface with `isToggleButtonVisible`

### Related PRs:
- EchoFramework: https://github.com/equinor/EchoFramework/pull/118
- EchopediaWeb: https://github.com/equinor/EchopediaWeb/pull/1067

### CODE REVIEW

**Comment by khollund** [comment]
Just a note, any additional planning on this should wait for the approved UX for new panel layout/ responsive design. As the logic of this will change 99% sure.

File: packages/echo-core/src/types/panel.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export interface Panel {
     key: string;
     label: string;
     icon: Icon;
+    isToggleButtonVisible?: (() => boolean) | boolean;
```
Comment (OveBDev) [inline packages/echo-core/src/types/panel.ts:21]: I think this should only be a function. Both a function and a boolean makes it hard to use and understand for other modules.

Comment (CsabaTheKiss) [reply packages/echo-core/src/types/panel.ts:21]: When I used this new property in echopedia with type Panel, VSCode provided enough hint to see that this is either a boolean or a function returning a boolean value:
image

Based on this I think it's easy to use and understand what is going on, when registering panels.

My reason to make it possible to pass just a primitive value is performance based: if one passes a function, that function will be called many times during UI changes & re-rendering. Calling a function is more costly then just checking a primitive value.
So I wanted to leave a possibility to pass just a primitive value - if that is enough in a given scenario - to improve performance of the app.

Comment (OveBDev) [reply packages/echo-core/src/types/panel.ts:21]: This is [premature optimization code smell](https://www.google.com/search?q=code+smell+premature+optimization&rlz=1C1GCEA_enNO867NO867&oq=code+smell+premature+optimization&aqs=chrome..69i57j69i64.4286j0j7&sourceid=chrome&ie=UTF-8)

Having to check if it's a function or boolean complicates the code and is not worth it.

**Pull request closed without merging**